moabb/datasets/preprocessing.py

```diff
@@ -218,9 +221,36 @@ def transform(self, X, y=None):
         if self.channels is None:
             picks = mne.pick_types(raw.info, eeg=True, stim=False)
         else:
+            available_channels = raw.info["ch_names"]
+            if self.interpolate_missing_channels:
+                missing_channels = list(set(self.channels).difference(available_channels))
+
+                # add missing channels (contains only zeros by default)
+                try:
+                    raw.add_reference_channels(missing_channels)
+                except IndexError:
+                    # Index error can occurs if the channels we add are not part of this epoch montage
+                    # Then log a warning
+                    montage = raw.info["dig"]
+                    warn(
+                        f"Montage disabled as one of these channels, {missing_channels}, is not part of the montage {montage}"
+                    )
+                    # and disable the montage
+                    raw.info.pop("dig")
+                    # run again with montage disabled
+                    raw.add_reference_channels(missing_channels)
+
+                # Trick: mark these channels as bad
+                raw.info["bads"].extend(missing_channels)
+                # ...and use mne bad channel interpolation to generate the value of the missing channels
+                raw.interpolate_bads(origin=(0, 0, 0.04))
```

Comment:
Hi @gcattan , I really like your idea about channel interpolation to match electrodes of two datasets.
Is there a reason why in interpolate_bads you have selected origin (0, 0, 0.04) ? Option "auto" will fit a sphere to headshape based on the existing electrodes and determine its radius and origin.

Reply:
Hi! I am not sure it will work if the montage is disabled. But let me try.

Reply:
Thanks @gcattan ! Yes, it doesn't seem that it fails due to this PR.

Reply:
All checks passed. @Sara04 FYI

Reply:
Thank you, @gcattan  !

Reply:
For me it is OK. @sylvchev , @bruAristimunha do you have some other comments?